docs/adr/adr-007-incentivized-testnet.md

```diff
@@ -0,0 +1,100 @@
+restart the node in order to continue sampling. Instead, there should be retry logic and potentially the graceful 
+shutdown of `DASer` and potentially node.
+
+### Public API 
+The public API is currently being drafted [here](https://github.com/celestiaorg/celestia-node/pull/506/files). There are
+several components that will go into implementing the overall design as well as the individual module implementations.
+
+### [Bad Encoding Fraud Proof](https://github.com/celestiaorg/celestia-node/issues/528) (continuation)
+The full implementation of bad encoding fraud proof (BEFP) generation, propagation and handling is required for the 
+incentivized testnet. At the moment, generation and broadcasting of BEFPs is still in review. 
+
+### [Testing](https://github.com/celestiaorg/celestia-node/issues/7)
+One of the biggest priorities for incentivized testnet is to test a lot of the assumptions we are working under in terms
+of sampling in realistic network conditions with a variety of topologies.
+
+Among the most important test cases, the following are required: 
+* [Large Block Tests](https://github.com/celestiaorg/celestia-node/issues/602)
+* [Block Recovery Tests](https://github.com/celestiaorg/test-infra/issues/21)
+* [Peer Discovery Tests](https://github.com/celestiaorg/celestia-node/issues/649)
+
+In terms of testing infrastructure, [testground](https://github.com/testground/testground) will be considered. A [proof of concept](https://github.com/celestiaorg/celestia-node/issues/638) will be done in order to accomplish the above tests.
+
+### [Wallet Management](https://github.com/celestiaorg/celestia-node/issues/415)
+All DA node types will support sending transactions and querying state-related information, meaning DA nodes 
+will need an account. So far, the standard key utility from cosmos-SDK has been made available via CLI for all node types,
+however more robust key/wallet management via the node's public API is required.
+
+### [`HeaderExchange` extensions and hardening](https://github.com/celestiaorg/celestia-node/issues/497)
+The current implementation of `HeaderExchange` has to be improved and hardened as there is a hard dependency on trusted 
+peers at the moment. Some high-level improvement requirements are: 
+
+* Read/write deadlines on libp2p streams
+* Server-side request limits
+* Server-side maximum request size configuration
+* Adjacency of a batched `ExtendedHeader` response must be verified in `HeaderEx` upon receiving the response
+* Request retries
+* Cycling peers for requests
+
+More detail can be found in the issue linked in the header.
+
+### [IPLD v1 migration + NMT/IPLD optimization](https://github.com/celestiaorg/celestia-node/issues/614)
+@Wondertan todo
+
+### [Telemetry](https://github.com/celestiaorg/celestia-node/issues/260)
+All node types must have metrics collection and tracing of hot paths implemented.
+
+## Nice to have
+### DB optimization
```

Comment:
Let's call it disk usage/storage optimizations. I doubt we will do any DB engine performance optimizations(besides updating Badger)